Source/WebCore/platform/graphics/cg/PathCG.cpp

```diff
@@ -21,167 +21,316 @@
  * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
  * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
 #include "config.h"
-#include "Path.h"
+#include "PathCG.h"
 
 #if USE(CG)
 
-#include "AffineTransform.h"
-#include "FloatRect.h"
 #include "GraphicsContextCG.h"
-#include "IntRect.h"
-#include <pal/spi/cg/CoreGraphicsSPI.h>
-#include <wtf/MathExtras.h>
+#include <wtf/NeverDestroyed.h>
 #include <wtf/RetainPtr.h>
-#include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
-static size_t putBytesNowhere(void*, const void*, size_t count)
+std::unique_ptr<PathCG> PathCG::create()
 {
-    return count;
+    return makeUnique<PathCG>();
 }
 
-static RetainPtr<CGContextRef> createScratchContext()
+std::unique_ptr<PathCG> PathCG::create(const PathStream& stream)
 {
-    CGDataConsumerCallbacks callbacks = { putBytesNowhere, 0 };
-    auto consumer = adoptCF(CGDataConsumerCreate(0, &callbacks));
-    auto context = adoptCF(CGPDFContextCreate(consumer.get(), 0, 0));
+    auto pathCG = PathCG::create();
+    if (!pathCG)
+        return nullptr;
```

Comment:
`pathCG` can never be null. If we change all these create functions to use `UniqueRef` that will be clearer.

Reply:
Done.